cloudfunctions2_basic/main.tf

```diff
@@ -1,4 +1,19 @@
 # [START functions_v2_basic]
+data "google_project" "project" {
+  provider = google-beta
+}
+
+resource "google_service_account" "account" {
+  account_id = "gcf-sa"
+  display_name = "Test Service Account"
+}
+
+resource "google_project_iam_member" "bucket" { 
+  project = data.google_project.project.project_id
+  role    = "roles/storage.objectViewer"
+  member  = "serviceAccount:${google_service_account.account.email}"
+}
```

Comment:
Do we use the `google_service_account` resource anywhere? If this was meant for the object access error, that is likely due to the Cloud Build Service Agent not having the permissions immediately on project creation which causes the build of the function to fail. https://github.com/terraform-google-modules/terraform-docs-samples/pull/181 should retry those cases.